packages/react-badge/src/BadgeBestPractices.md

```diff
@@ -10,12 +10,37 @@
 
 ### Screen Readers
 
-- Badges should be given a meaningful description. This description will be applied, via “aria-describedby” to the element decorated by the Badge. For example, "Chat, 6 unread" or similar.
-  General guidance is that the badge icon is marked as “aria-hidden” by default.
+Badge content is exposed as text, and is treated by screen readers as if it were inline content of the control it is associated with. This should provide a reasonable default for most badges that contain plain text, such as the `CounterBadge`.
+
+There are two actions authors should consider taking when using Badge to improve this experience:
+
+1. If the badge contains a custom icon, that icon must be given alternative text with `aria-label`, unless it is purely presentational:
+
+```jsx
+<Badge icon={<PasteIcon aria-label="paste" />} />
+```
+
+2. If the text of the badge itself is not sufficient to convey its meaning, it can either be given additional hidden text, or the parent element given an explicit label:
+
+```jsx
+<button>
+  Inbox
+  <Badge>6<span class="visuallyHidden"> unread messages</span></Badge>
```

Comment:
during Badge design review it was agreed to put additional information to the description, can you please elaborate on reasons for changing the recommendation?

Reply:
maybe something changed so it can be worth re-testing but I remember bugs where hidden texts were causing issues with voice control / dictation - as the voice control software required to read the whole label in order to navigate to the element. Hiding works very well for screen readers, but it might not be the case for all scenarios :(

Reply:
Did a quick test:

```html
<html>

<head>
  <style>
    .visually-hidden {
      clip: rect(0 0 0 0);
      clip-path: inset(50%);
      height: 1px;
      overflow: hidden;
      position: absolute;
      white-space: nowrap;
      width: 1px;
    }
  </style>
</head>

<body>
  <button>Batman</button>
  <button>Strawberry <span class="visually-hidden">superhero</span></button>
</body>
```

I can use Voice Control to click on Batman, but Strawberry is not being recognized.

Reply:
@jurokapsiar which voice control software are you using? Partial labels have not been a problem in the past, at least with the voice control software I'm familiar with. If it is now, that seems like a major issue for a lot of common approaches to accessible text, and probably something we should follow up on as a bug with the software itself.

The issue with putting badge info as a description mostly falls into two categories:
1. The description needs to go on something that accepts an accessible name/description. It's worth noting that this did not happen in a number of our existing examples, meaning the description was effectively worthless.
2. A description is much less likely to be read, since it depends heavily on the user having their verbosity relatively high (or specifically turning on descriptions). Generally the accessible description should be used for information that is fully optional/unnecessary for the user to receive. Sometimes badges will fall into this category, hence the best practices text on how to mark those up. I just don't think it should be our default.

The first issue is the much more serious one. Based on the variety of places a badge can be used, the only real safe way to default to exposing the text is to have it show up as actual content.